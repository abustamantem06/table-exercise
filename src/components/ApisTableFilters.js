import { Form, Row, Col, Button } from "react-bootstrap"
import { AUTH_FILTER_OPTIONS, CORS_FILTER_OPTIONS, HTTPS_FILTER_OPTIONS } from "../utils/constants";

const ApisTableFilters = ({onAuthFilter, onHttpsFilter, onCorsFilter, onUpdateFilters, onNameFilter, onDescriptionFilter, onCategoryFilter}) => {

    const updateData = e => {
        e.preventDefault();
        onUpdateFilters();
    }

    return (
        <div className="p-2">
            <Form>
                <Row className="py-2">
                    {/* AUTH */}
                    <Col>
                        <Form.Group>
                            <Form.Label>Auth</Form.Label>
                                <Form.Select onChange={e => onAuthFilter(e.target.value)}>
                                    <option></option>
                                    <option value={AUTH_FILTER_OPTIONS.API_KEY}>{AUTH_FILTER_OPTIONS.API_KEY}</option>
                                    <option value={AUTH_FILTER_OPTIONS.NULL}>{AUTH_FILTER_OPTIONS.NULL}</option>
                                </Form.Select>
                            </Form.Group>
                    </Col>

                    {/* HTTPS */}

                    <Col>
                        <Form.Group>
                            <Form.Label>HTTPS</Form.Label>
                                <Form.Select onChange={e => onHttpsFilter(e.target.value)}>
                                    <option></option>
                                    <option value={HTTPS_FILTER_OPTIONS.TRUE}>{HTTPS_FILTER_OPTIONS.TRUE}</option>
                                    <option value={HTTPS_FILTER_OPTIONS.FALSE}>{HTTPS_FILTER_OPTIONS.FALSE}</option>
                                </Form.Select>
                            </Form.Group>
                    </Col>

                    {/* CORS */}
                    <Col>
                        <Form.Group>
                            <Form.Label>CORS</Form.Label>
                                <Form.Select onChange={e => onCorsFilter(e.target.value)}>
                                    <option></option>
                                    <option value={CORS_FILTER_OPTIONS.YES}>{CORS_FILTER_OPTIONS.YES}</option>
                                    <option value={CORS_FILTER_OPTIONS.NO}>{CORS_FILTER_OPTIONS.NO}</option>
                                    <option value={CORS_FILTER_OPTIONS.UNKNOWN}>{CORS_FILTER_OPTIONS.UNKNOWN}</option>
                                </Form.Select>
                            </Form.Group>
                    </Col>

                    {/* Name */}
                    <Col>
                        <Form.Label>Name:</Form.Label>
                        <Form.Control type="text" onChange={e => onNameFilter(e.target.value)}/>
                    </Col>

                    {/* Description */}
                    <Col>
                        <Form.Label>Description:</Form.Label>
                        <Form.Control type="text" onChange={e => onDescriptionFilter(e.target.value)}/>
                    </Col>

                    {/* Category */}
                    <Col>
                        <Form.Label>Category:</Form.Label>
                        <Form.Control type="text" onChange={e => onCategoryFilter(e.target.value)}/>
                    </Col>
                    
                </Row>
                <Row className="py-2">
                    <Col>
                        <Button className="btn yellow-bg" type="submit" onClick={e => updateData(e)}>Update</Button>
                    </Col>
                </Row>
            </Form>
        </div>
        
    );
}

export default ApisTableFilters;