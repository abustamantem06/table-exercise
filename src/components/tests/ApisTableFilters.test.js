import renderer from 'react-test-renderer';
import ApisTableFilters from '../ApisTableFilters';

describe('ApiTableFilters', () => {
    it('renders correctly', () => {
        const tree = renderer
        .create(<ApisTableFilters />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});