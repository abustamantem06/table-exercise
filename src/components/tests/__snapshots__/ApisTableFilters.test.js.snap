// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiTableFilters renders correctly 1`] = `
<div
  className="p-2"
>
  <form
    className=""
  >
    <div
      className="py-2 row"
    >
      <div
        className="col"
      >
        <div>
          <label
            className="form-label"
          >
            Auth
          </label>
          <select
            className="form-select"
            onChange={[Function]}
          >
            <option />
            <option
              value="apiKey"
            >
              apiKey
            </option>
            <option
              value="null"
            >
              null
            </option>
          </select>
        </div>
      </div>
      <div
        className="col"
      >
        <div>
          <label
            className="form-label"
          >
            HTTPS
          </label>
          <select
            className="form-select"
            onChange={[Function]}
          >
            <option />
            <option
              value="true"
            >
              true
            </option>
            <option
              value="false"
            >
              false
            </option>
          </select>
        </div>
      </div>
      <div
        className="col"
      >
        <div>
          <label
            className="form-label"
          >
            CORS
          </label>
          <select
            className="form-select"
            onChange={[Function]}
          >
            <option />
            <option
              value="yes"
            >
              yes
            </option>
            <option
              value="no"
            >
              no
            </option>
            <option
              value="unknown"
            >
              unknown
            </option>
          </select>
        </div>
      </div>
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Name:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Description:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Category:
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
        />
      </div>
    </div>
    <div
      className="py-2 row"
    >
      <div
        className="col"
      >
        <button
          className="btn yellow-bg btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Update
        </button>
      </div>
    </div>
  </form>
</div>
`;
